### PR TITLE
USM: tests: fix TLS+Redis classification test

### PR DESCRIPTION
**Pull request by Yumasi**
### What does this PR do?

Fix a test that was wrongly using a plaintext connection in a TLS test, causing tests failures.

### CODE REVIEW

File: pkg/network/tracer/tracer_usm_linux_test.go [hunk 1]
```diff
@@ -1741,9 +1741,20 @@ func testRedisProtocolClassificationInner(t *testing.T, tr *Tracer, clientHost,
...
+				dialContext := defaultDialer.DialContext
+				if withTLS {
+					tlsDialer := &tls.Dialer{
+						NetDialer: defaultDialer,
+						Config: &tls.Config{
+							InsecureSkipVerify: true,
+						},
+					}
+					dialContext = tlsDialer.DialContext
+				}
```
Comment (guyarb) [inline pkg/network/tracer/tracer_usm_linux_test.go:1753]: why only for this test?

Comment (Yumasi) [reply pkg/network/tracer/tracer_usm_linux_test.go:1753]: The Redis client wrapper we use in the other tests already does this.

Comment (guyarb) [reply pkg/network/tracer/tracer_usm_linux_test.go:1753]: let's add an inline comment in the code
that test is not using the client to create an actual protocol error, therefore we need ....

**Review by guyarb — APPROVED** [review]

**Comment by guyarb** [comment]
the tests are still flaky - https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/543385813

File: .gitlab/kernel_matrix_testing/common.yml [hunk 2]
```diff
@@ -226,10 +226,13 @@
     - NESTED_VM_CMD="/home/ubuntu/connector -host ${MICRO_VM_IP} -user root -ssh-file /home/kernel-version-testing/ddvm_rsa -vm-cmd 'CI=true /root/fetch_dependencies.sh ${ARCH} && /opt/micro-vm-init.sh -test-tools /opt/testing-tools -retry ${RETRY} -test-root /opt/${TEST_COMPONENT}-tests -packages-run-config /opt/${TEST_SET}.json'"
     - $CI_PROJECT_DIR/connector-$ARCH -host $INSTANCE_IP -user ubuntu -ssh-file $AWS_EC2_SSH_KEY_FILE -vm-cmd "${NESTED_VM_CMD}"
     - ssh metal_instance "ssh ${MICRO_VM_IP} '/opt/testing-tools/test-json-review'"
+    - ssh metal_instance "scp -r ${MICRO_VM_IP}:/tmp/test_pcaps /home/ubuntu/test_pcaps-${ARCH}-${TAG}-${TEST_SET}"
```
Comment (brycekahle) [inline .gitlab/kernel_matrix_testing/common.yml:229]: I think you might want this in `collect_outcomes_kmt` so they are collected regardless of test success/failure.

**Pull request closed without merging**